src/include/ndpi_main.h

```diff
@@ -107,11 +107,6 @@ extern "C" {
   extern void ndpi_set_proto_subprotocols(struct ndpi_detection_module_struct *ndpi_mod,
 				      int protoId, ...);
 
-  extern int ndpi_packet_src_ip_eql(const struct ndpi_packet_struct *packet, const ndpi_ip_addr_t * ip);
```

Comment:
Those functions could still be useful.
What about:

```C
extern int ndpi_packet_src_ip_eql(const ndpi_ip_addr_t * a, const ndpi_ip_addr_t * b);
```

Reply:
I think the entire `ndpi_ip_addr_t` is completely bogus...  because we are not able to tell ipv4 from ipv6 only from such object, if we use prefixes and not addresses only
The function
```
u_int8_t ndpi_is_ipv6(const ndpi_ip_addr_t *ip) {
  return(ip->ipv6.u6_addr.u6_addr32[1] != 0 || ip->ipv6.u6_addr.u6_addr32[2] != 0 ||
	 ip->ipv6.u6_addr.u6_addr32[3] != 0);
}
```
is wrong in these conditions. (trivial example: fec0::/8 will be classified as ipv4...)
The only usage of `ndpi_ip_addr_t` in the library is via `ndpi_ptree_insert()` with does handles prefixes so it is potentially bogus too. Fortunately:
1) `ndpi_ptree_insert()` is never used by the library
2) it is used by ntopng, but with full address, i.e. without any issues
```
ivan@ivan-Latitude-E6540:~/svnrepos/ntopng(dev)$ git grep ndpi_ptree_insert
src/flow_checks/UnexpectedServer.cpp:            ndpi_ptree_insert(whitelist_ptree, &a, 32, naddr);
src/flow_checks/UnexpectedServer.cpp:            ndpi_ptree_insert(whitelist_ptree, &a, 128, naddr);

```

Bottom line: I think we can avoid adding (back) something like that right now
```
extern int ndpi_packet_src_ip_eql(const ndpi_ip_addr_t * a, const ndpi_ip_addr_t * b);
```
and in the future we need to rework all the `ndpi_ip_addr_t` stuff (remove it completely? Handle only addresses and not prefixes?)

What do you think?